presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -868,57 +870,38 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
+        Geometry geometry = deserialize(input);
         validateType("ST_Points", geometry, EnumSet.of(LINE_STRING, POLYGON, POINT, MULTI_POINT,
                 MULTI_LINE_STRING, MULTI_POLYGON, GEOMETRY_COLLECTION));
         if (geometry.isEmpty()) {
             return null;
         }
 
-        int pointCount = countNumPoints(geometry);
+        int pointCount = geometry.getNumPoints();
         BlockBuilder blockBuilder = GEOMETRY.createBlockBuilder(null, pointCount);
         buildPointsBlock(geometry, blockBuilder);
-        return blockBuilder.build();
-    }
-
-    private static int countNumPoints(OGCGeometry geometry)
-    {
-        if (geometry == null) {
-            return 0;
-        }
 
-        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
-        int pointCount = 0;
-        if (type == GeometryType.GEOMETRY_COLLECTION) {
-            OGCGeometryCollection collection = (OGCGeometryCollection) geometry;
-            for (int i = 0; i < collection.numGeometries(); i++) {
-                OGCGeometry entry = collection.geometryN(i);
-                pointCount += countNumPoints(entry);
-            }
-        }
-        else {
-            pointCount = getPointCount(geometry);
-        }
-        return pointCount;
+        return blockBuilder.build();
     }
 
-    private static void buildPointsBlock(OGCGeometry geometry, BlockBuilder blockBuilder)
+    private static void buildPointsBlock(Geometry geometry, BlockBuilder blockBuilder)
     {
-        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.getGeometryType());
         if (type == GeometryType.POINT) {
-            GEOMETRY.writeSlice(blockBuilder, serialize(createFromEsriGeometry(geometry.getEsriGeometry(), null)));
+            GEOMETRY.writeSlice(blockBuilder, serialize(geometry));
         }
         else if (type == GeometryType.GEOMETRY_COLLECTION) {
-            OGCGeometryCollection collection = (OGCGeometryCollection) geometry;
-            for (int i = 0; i < collection.numGeometries(); i++) {
-                OGCGeometry entry = collection.geometryN(i);
+            GeometryCollection collection = (GeometryCollection) geometry;
+            for (int i = 0; i < collection.getNumGeometries(); i++) {
+                Geometry entry = collection.getGeometryN(i);
                 buildPointsBlock(entry, blockBuilder);
             }
         }
         else {
-            MultiVertexGeometry vertices = (MultiVertexGeometry) geometry.getEsriGeometry();
-            for (int i = 0; i < vertices.getPointCount(); i++) {
-                GEOMETRY.writeSlice(blockBuilder, serialize(createFromEsriGeometry(vertices.getPoint(i), null)));
+            GeometryFactory gFactory = geometry.getFactory();
```

Comment:
Instead of abbreviating, use `geometryFactory`.

Reply:
Done.